stan/math/prim/fun/ones_array.hpp

```diff
@@ -0,0 +1,25 @@
+#ifndef STAN_MATH_PRIM_FUN_ONES_ARRAY_HPP
+#define STAN_MATH_PRIM_FUN_ONES_ARRAY_HPP
+
+#include <stan/math/prim/err.hpp>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return an array of ones
+ *
+ * @param K size of the array
+ * @return An array of size K with all elements initialised to 1.
+ * @throw std::domain_error if K is negative.
+ */
+inline std::vector<double> ones_array(int K) {
+  check_nonnegative("ones_array", "size", K);
+  return std::vector<double>(K, 1);
```

Comment:
This can just be `return {K, 1};`  You can always return curly braces around constructor args for the return type.

And same for next definition.

Reply:
This doesn't work:
```
./stan/math/prim/fun/ones_array.hpp:19:11: error: non-constant-expression cannot be
      narrowed from type 'int' to 'double' in initializer list [-Wc++11-narrowing]
  return {K, 1};
          ^
./stan/math/prim/fun/ones_array.hpp:19:11: note: insert an explicit cast to silence
      this issue
  return {K, 1};
          ^
          static_cast<double>( )
```
If I use the static cast around `K`, it builds a vector of two elements.

Reply:
Thanks for trying.  I've never seen that error and it's suprrising in that I thought you could always replace return-type constructor calls with braces.   It clearly requires a stricter match.

Reply:
It does look like declaring `const int K` might also work, but I'd rather not sprinkle constants around.